### ISSUE TITLE
@DataSet not detect when annotated with @BeforeAll up the parent class hierarchy

### ISSUE THREAD

**Issue by hosing-lee** [database-rider/database-rider#610]
DbRider versión 1.44.0

In the class `DBUnitExtension`  the method:

```
    private Set<Method> findCallbackMethods(Class testClass, Class callback) {
        final Set<Method> methods = new HashSet<>();
        Stream.of(testClass.getSuperclass()
                        .getDeclaredMethods(), testClass.getDeclaredMethods())
                .flatMap(Stream::of)
                .filter(m -> m.getAnnotation(callback) != null)
                .forEach(m -> methods.add((Method) m)); //do not use Collectors.toSet here: stream incompatible types
        return Collections.unmodifiableSet(methods);
    }
```

only finds @BeforeAll annotation on the class itself and the parent class. But if the parent class is inheriting another class with @BeforeAll annotation it is not detected.

Can this be fixed to detect all parent classes?

In version 1.26.0 it works, it uses`testClass.getSuperclass().getMethods()` is used instead of `getDeclaredMethods`.

**Comment by hosing-lee** [comment]
I see the fix was related to https://github.com/database-rider/database-rider/issues/331#issuecomment-890307713

With that fix, only the immediate supper class method can be detected and not further up the hierarchy.

**Comment by rmpestano** [comment]
Fixed by #612, will be available in 1.45.0, to be released soon
